### PR TITLE
Prototyping WES support

### PR DESCRIPTION
**Pull request by briandoconnor**
This isn't totally ready to go but I'm making the pull request so @achave11 can compare and contrast to his work, leave comments, make tickets, etc.  

The goal here was to just layer on GA4GH WES support to Consonance.  The basic responses are working however it hasn't actually been tested for real workflows yet running on provisioned worker nodes.  Also, there's a bunch of features/requirements in WES responses that we currently don't propagate back via Consonance (output files for example).  We're going to need to work with the Dockstore CLI developers to implement a mechanism to get this data back.

Adding @denis-yuen to this.  @achave11 we should probably complete a review first and then ping Denis to take a look if he's interested/has bandwidth.

### CODE REVIEW

File: consonance-webservice/src/main/java/io/swagger/workflow/api/impl/Ga4ghApiServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+
+            // engine
+            Map<String, String> engineMap = new HashMap<String, String>();
+            // FIXME: these need to be coordinated with the versions installed by the playbook for the worker
+            engineMap.put("Cromwell", "29");
+            engineMap.put("cwltool", "1.0.20170828135420");
+            serviceInfo.setEngineVersions(engineMap);
+
+            // FIXME: need to confirm this list with the Dockstore CLI
+            serviceInfo.setSupportedFilesystemProtocols(Arrays.asList("http", "https", "s3", "file"));
```
Comment (denis-yuen) [inline consonance-webservice/src/main/java/io/swagger/workflow/api/impl/Ga4ghApiServiceImpl.java:81]: Yeah, you can probably do some parsing of `dockstore plugin list` or create a PR for it to return something more machine-readible when running with something like `dockstore --script plugin list`

Comment (denis-yuen) [reply consonance-webservice/src/main/java/io/swagger/workflow/api/impl/Ga4ghApiServiceImpl.java:81]: There are also the default protocols provided via vfs2 https://github.com/ga4gh/dockstore/blob/3083a8769f9096a2112c2f175de538afd70c0ba3/dockstore-client/src/main/java/io/dockstore/common/FileProvisioning.java#L388

**Review by denis-yuen — COMMENTED** [review]
that's a lot of generated code. 
FYI, we stopped using dependencyci in our projects, too many false positives

File: consonance-webservice/src/main/java/io/swagger/workflow/api/impl/Ga4ghApiServiceImpl.java (hunk 1 above)
Comment (briandoconnor) [reply consonance-webservice/src/main/java/io/swagger/workflow/api/impl/Ga4ghApiServiceImpl.java:81]: Thanks Denis.  You, Charles, and I should chat about making the dockstore CLI more friendly for WES to be built on top of.  I have some ideas.

**Comment by briandoconnor** [comment]
The latest commit is a work in progress.  @achave11 please take a look at the README and see if you can finish the work of upgrading the various dependencies.  I'm trying to get this up to the current dockstore release but it's quite the ripple effect in terms of versions changed and the number of exclusions that need to be done to resolve version conflicts.

File: consonance-arch/src/test/java/io/consonance/arch/test/TestWorkerWithMocking.java [hunk 2]
```diff
@@ -1,281 +0,0 @@
-/*
```
Comment (briandoconnor) [inline consonance-arch/src/test/java/io/consonance/arch/test/TestWorkerWithMocking.java:1]: This class will come back, just removed it to see if I could sidestep the classloader issue I ran into

**Comment by achave11-ucsc** [comment]
Handled in #161, closing this PR.

**Pull request closed without merging**